Study definition

ZoeMZou opened the pull request:
This PR includes the following updates:

**1. Created practice-level measure tables** (n = 12 GP characteristics and secondary care use indicators) across three cohorts (prcovid, postcovid1, postcovid2), resulting in **36 measure tables** saved under `/output/measures`.

**2. Generated patient-level datasets for each of the three cohorts**, including:

- Diagnosis indicators for the 20 conditions listed in the Cambridge Multimorbidity Score,

- COVID-19 vaccination dates, and

- Eligibility dates for COVID-19 vaccination (based on JCVI group).

This results in **3 datasets** saved under `/output/dataset_definition`.

**3. Created a study date dataset used for calculating JCVI group and vaccine eligibility dates** (**1 file** under `/output/dataset_definition`).

**Total outputs: 40 files.**

Shrinkhala-D commented on analysis/dataset_definition/measures_cohorts.py:
```diff
@@ -0,0 +1,292 @@
+            group_by={
+                "practice_pseudo_id": practice_id
+            },
+            intervals=months(1).starting_on(start_cohort),
```
For the cross-sectional exposure variables, why are we defining them as the proportion in the month of October instead of September? So our "exposure" has a one-month cross-over with our outcome?

ZoeMZou replied:
Thanks, Shrinkhala. This was discussed and resolved in our meeting on 10th April. Only the interval start date were used for calculating the numerator. The purpose of defining an interval—regardless of its length—is simply to enable use of the measure framework. Based on this, I’m going ahead and resolving this.

Shrinkhala-D commented on analysis/dataset_definition/measures_cohorts.py:
```diff
@@ -0,0 +1,292 @@
+            measures.define_measure(
+                name = "exp_count_consultation_m",
+                numerator = exp_num_consrate,
+                intervals = months(12).ending_on(start_cohort)
```
For neatness, can we modify this so the interval ends on Sept 30? E.g set the end date to: start_cohort-1? 
The consultation measures currently run from, e.g. Jan 2-Feb1, Feb 2 - March 1, etc. Would be neat to have them allign to the calendar month dates.

ZoeMZou replied:
Thanks Shrinkhala, they have been revised, also the newly added vax relevant variables, please see the revision as follows:
https://github.com/opensafely/WinterPressuresDescriptive/blob/f2d900b82c7f26df02ed7f6c0814157e25762716/analysis/dataset_definition/measures_cohorts.py#L258-L272

Shrinkhala-D commented on analysis/dataset_definition/variables_dataset.py:
```diff
@@ -0,0 +1,341 @@
+        otherwise="M"
+    )
+    exp_bin_smoker = (tmp_cat_smoking == "S")
+
```
I couldn't find where obesity was defined as a variable?

ZoeMZou replied:
They have now been added:
https://github.com/opensafely/WinterPressuresDescriptive/blob/f2d900b82c7f26df02ed7f6c0814157e25762716/analysis/dataset_definition/variables_dataset.py#L129-L139

https://github.com/opensafely/WinterPressuresDescriptive/blob/f2d900b82c7f26df02ed7f6c0814157e25762716/analysis/dataset_definition/measures_cohorts.py#L232-L237

ZoeMZou commented:
Hi @Shrinkhala-D,

Thank you for your review. I’ve now revised the code in response to your comments. In addition, I spotted a few issues and made the following updates:

1. Definition of vax_flu has been added—denominator is patients aged 65 or over, and the numerator is those who received a flu vaccine in the year prior to the winter months.
https://github.com/opensafely/WinterPressuresDescriptive/blob/f2d900b82c7f26df02ed7f6c0814157e25762716/analysis/dataset_definition/variables_measure.py#L38-L47
https://github.com/opensafely/WinterPressuresDescriptive/blob/f2d900b82c7f26df02ed7f6c0814157e25762716/analysis/dataset_definition/measures_cohorts.py#L265-L272

2. As we’re not examining during-COVID cohorts, defining vax_covid for the pre-pandemic period isn’t meaningful as no one had covid-vax before pandemic. It may be relevant for post-COVID1, but likely not for post-COVID2 or 3, as the pandemic phase has passed. As you may noticed, I’ve added each patients COVID eligibility date and vaccination date to the patient-level dataset previously, in case we decide to explore COVID vaccination in later analyses.

3. Proportion of patients with missing values for demographic variables has been added.

4. Fixed a small bug in the smoking codes.

5. Tidied the script and renamed some measures for improved readability, e.g. using the format of `exp_prop_{var}` to indicate proportion variables etc. 

Let me know if you spot anything else! Happy to discuss. Thanks again.

Shrinkhala-D commented on analysis/dataset_definition/variables_vax_covid.py:
```diff
@@ -0,0 +1,473 @@
+    when(vax_jcvi_age_1 >= 80).then("02"),
+    when(vax_jcvi_age_1 >= 75).then("03"),
+    when((vax_jcvi_age_1 >= 70) | 
+         (cev_group & (vax_jcvi_age_1 >= 16) & (~preg_group))).then("04"),
```
Could we chat through the logic for vax_cat_jcvi_group = 04? Because it includes everyone over 70 OR everyone who's over 16, severely clinically vulnerable AND pregnant?

Shrinkhala-D reviewed (commented):
The code is looking amazing Zoe! No huge comments - can we touch base about the vaccine definitions?

ZoeMZou commented on analysis/dataset_definition/measures_cohorts.py:
```diff
@@ -0,0 +1,334 @@
+                    numerator = measures_vax[measure],
+                    denominator = inex_bin_reg_cs & inex_bin_alive & (exp_num_age >= 65),
+                    intervals = years(1).ending_on(start_cohort - days(1))
+                )
```
Also we could chat about the age group <5, as suggested by Rachel

Shrinkhala-D replied:
Ok to summarise our chat - the denom will be those >=65, those <5, and people who were pregnant
